src/TieBreakingMarginComparator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+namespace PivotLibre\Tideman;
+
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+
+class TieBreakingMarginComparator
+{
+
+    use LoggerAwareTrait;
+
+    private $tieBreaker;
+
+    public function __construct(MarginTieBreaker $tieBreaker)
+    {
+        $this->tieBreaker = $tieBreaker;
+    }
+
+    /**
+     * Compares two Margins, returning an integer to indicate their relative ordering.
+     * @param Margin $a
+     * @param Margin $b
+     * @return an int :
+     *  A positive int if Margin $a is more preferred than Margin $b
+     *  A negative int if Margin $b is more preferred than Margin $a
+     *
+     * This function should never return zero. In the event of a tie, it should use the TieBreaker specified in the
+     * constructor to determine a nonzero integer indicating which Margin should be treated as though it were more
+     * preferred than the other.
+     */
+    public function compare(Margin $a, Margin $b) : int
+    {
+        $differenceOfStrength = $a->getDifference() - $b->getDifference();
+        if (0 == $differenceOfStrength) {
+            // $this->logger->info("Tie between two Margins:\n$a\n$b\n");
```

Comment:
@andrewshell I kept running into `logger` being null. Do we need to inject loggers in every class, or is the LoggerAwareTrait supposed to do that for us? If the latter, is there some configuration we need to tweak? I'm new to traits in general.